File: app/src/main/java/com/mx/gillustrated/listener/ListenerListViewScrollHandler.java
package com.mx.gillustrated.listener;

import com.mx.gillustrated.R;

import android.util.Log;
import android.view.View;
import android.widget.AbsListView;
import android.widget.AbsListView.OnScrollListener;
import android.widget.ListView;
import android.widget.RelativeLayout;
import android.widget.TextView;

public class ListenerListViewScrollHandler implements OnScrollListener {
	
	private boolean isLastRow = false;
    private int lastCount = 0; //避免反复触发last事件
	private ListView deliverylist;
	private RelativeLayout pageVboxLayout; //计数用   例： 1/4
	private TextView pageText;
	private int mAddRowNum = 0;//增加了几个额外行 刷新行或者表头行都计入数量
	private ScrollHandle mScrollHandle;

	public ListenerListViewScrollHandler(ListView lv, RelativeLayout rl, int addRowNum, ScrollHandle scrollHandle)
	{
		mAddRowNum = addRowNum;
		mScrollHandle = scrollHandle;
		init(lv, rl);
	}

	public ListenerListViewScrollHandler(ListView lv, RelativeLayout rl)
	{
		init(lv, rl);
	}

	private void init(ListView lv, RelativeLayout rl){
		deliverylist = lv;
		pageVboxLayout = rl;
		pageText = (TextView) pageVboxLayout.findViewById(R.id.pageText);
	}

	@Override
	public void onScroll(AbsListView view, int firstVisibleItem,
			int visibleItemCount, int totalItemCount) {
        if(firstVisibleItem == 0)
            lastCount = 0;

		if (pageVboxLayout.getVisibility() == View.VISIBLE) {
			setPageBox(true);
		}
		if (firstVisibleItem + visibleItemCount == totalItemCount
				&& totalItemCount > 0 && lastCount < totalItemCount) {
            isLastRow = true;
            lastCount = totalItemCount;
		}
	}

	private void setPageBox(boolean isVisible){
		int totalCount = deliverylist.getCount() - mAddRowNum;
		int visibleCount = deliverylist.getLastVisiblePosition() + 1 - mAddRowNum;
		pageText.setText( isVisible ? visibleCount + "/" + totalCount : "");
		pageVboxLayout.setVisibility(isVisible ? View.VISIBLE : View.GONE);
	}

	@Override
	public void onScrollStateChanged(AbsListView view, int scrollState) {
		
		if (scrollState == AbsListView.OnScrollListener.SCROLL_STATE_TOUCH_SCROLL) {
			setPageBox(true);
		}
		if (scrollState == AbsListView.OnScrollListener.SCROLL_STATE_FLING) {
			setPageBox(true);
		}
		if (scrollState == AbsListView.OnScrollListener.SCROLL_STATE_IDLE) {
			setPageBox(false);
			if(isLastRow && mScrollHandle != null) {
				mScrollHandle.scrollLastRow(deliverylist.getCount() - mAddRowNum);
				isLastRow = false;
			}
		}
	}

	public interface ScrollHandle{
		void scrollLastRow(int totalItemCount);
	}
	 

}

File: app/src/main/java/com/mx/gillustrated/listener/ListenerListViewScrollHandler.kt
package com.mx.gillustrated.listener

import com.mx.gillustrated.R
import android.view.View
import android.widget.AbsListView
import android.widget.AbsListView.OnScrollListener
import android.widget.ListView
import android.widget.RelativeLayout
import android.widget.TextView

class ListenerListViewScrollHandler constructor(lv: ListView, rl: RelativeLayout,
                                                private val mAddRowNum: Int, //增加了几个额外行 刷新行或者表头行都计入数量
                                                private val mScrollHandle: ScrollHandle?) : OnScrollListener {

    private var isLastRow = false
    private var lastCount = 0 //避免反复触发last事件
    private lateinit var listView: ListView
    private lateinit var pageVBoxLayout: RelativeLayout //计数用   例： 1/4
    private lateinit var pageText: TextView

    init {
        init(lv, rl)
    }

    constructor(lv: ListView, rl: RelativeLayout): this(lv, rl, 0, null)

    private fun init(lv: ListView, rl: RelativeLayout) {
        listView = lv
        pageVBoxLayout = rl
        pageText = pageVBoxLayout.findViewById<View>(R.id.pageText) as TextView
    }

    override fun onScroll(view: AbsListView, firstVisibleItem: Int,
                          visibleItemCount: Int, totalItemCount: Int) {
        if (firstVisibleItem == 0)
            lastCount = 0

        if (pageVBoxLayout.visibility == View.VISIBLE) {
            setPageBox(true)
        }
        if (firstVisibleItem + visibleItemCount == totalItemCount
                && totalItemCount > 0 && lastCount < totalItemCount) {
            isLastRow = true
            lastCount = totalItemCount
        }
    }

    private fun setPageBox(isVisible: Boolean) {
        val totalCount = listView.count - mAddRowNum
        val visibleCount = listView.lastVisiblePosition + 1 - mAddRowNum
        pageText.text = if (isVisible) "$visibleCount/$totalCount" else ""
        pageVBoxLayout.visibility = if (isVisible) View.VISIBLE else View.GONE
    }

    override fun onScrollStateChanged(view: AbsListView, scrollState: Int) {

        if (scrollState == OnScrollListener.SCROLL_STATE_TOUCH_SCROLL) {
            setPageBox(true)
        }
        if (scrollState == OnScrollListener.SCROLL_STATE_FLING) {
            setPageBox(true)
        }
        if (scrollState == OnScrollListener.SCROLL_STATE_IDLE) {
            setPageBox(false)
            if (isLastRow && mScrollHandle != null) {
                mScrollHandle.scrollLastRow(listView.count - mAddRowNum)
                isLastRow = false
            }
        }
    }

    interface ScrollHandle {
        fun scrollLastRow(totalCount: Int)
    }

}
